book/src/hello_world.md

```diff
@@ -19,4 +19,61 @@ Now let's try for the invalid input `1`
 powdr pil hello_world.asm --field bn254 --force --inputs 1 --prove-with halo2
 ```
 
-We observe that witness generation fails, and no proof is created.
\ No newline at end of file
+We observe that witness generation fails, and no proof is created.
+
+# Setup & Verification
+
+The example above omits some important steps in proof generation: Setup and
+Verification key generation.  Some proof systems, such as Halo2 (and other
+SNARKs), require a Setup to be performed before the proof. Such Setup can be
+specific to the program or universal, where its artifact is a binary usually
+called `parameters` or `params`. STARKs do not require a previous Setup.
+
+Another step required before the proof is computed is key generation. A
+`proving key` and a `verification key` are generated taking into account the
+program, private inputs, and potentially Setup parameters. The `proving key` is
```

Comment:
What do you mean by "private inputs"?

How about "generated taking into account the constraints of the zk-VM, the program, and potentially Setup parameters"?

Reply:
Or maybe in this context only "constraints and potentially Setup parameters". Building a zk-VM with a program running on is a special case of that.

Reply:
actually the inputs thing is wrong here